Observer/SalesOrderPlaceAfterObserver.php

```diff
@@ -117,15 +119,22 @@ protected function fetchContractDetail($contractId)
         $response = $this->callCurl($url, $request);
 
         if (isset($response['response']) && isset($response['signature'])) {
+
             if ($this->validateResponse($response['response'], $response['signature'])) {
-                //$response['ConsumerCurrency']; to save currency info later on
-                if ($response && isset($response['Payment'])) {
+                //lack of this line was the problem; without this it was looking for desired elements at wrong place/depth
+                //in the nested data structure.
+                //Also converting string in $response['response'] to array so that the rest of the code would
+                //work with minimal changes.
+                $response_extracted = json_decode($response['response'], true);
+
+                if ($response_extracted  && isset($response_extracted['Payment'])) {
                     $detail=[];
-                    $detail['method']=$response['Payment']['Method'];
-                    $detail['identifier']=$response['Payment']['AccountIdentifier'];
+                    $detail['currency']= $response_extracted['ConsumerCurrency'];
+                    $detail['method']=$response_extracted['Payment']['Method'];
+                    $detail['identifier']=$response_extracted['Payment']['AccountIdentifier'];
 
-                    if (isset($response['Times']['Expiry'])) {
-                        $expredAt = explode('T', $response['Times']['Expiry']);
+                    if (isset($response_extracted['Times']['Expiry'])) {
+                        $expredAt = explode('T',$response_extracted['Times']['Expiry']);
```

Comment:
can you fix the expredAt typo here to "expiredAt"?

Reply:
Sure Lizzy. Here you go: 298f55e308cde2677e9a1d46ae7ccc0926cd6562